File: NetFramework/FlexGrid/CS/RowDetails/Samples/InputPanelSample.cs
using System;
using System.Data;
using System.Windows.Forms;
using C1.Win.FlexGrid;

namespace RowDetails.Samples
{
    public partial class InputPanelSample : UserControl
    {
        public InputPanelSample()
        {
            InitializeComponent();
        }

        private void InputPanel_Load(object sender, EventArgs e)
        {
            if (DesignMode) return;

            string conn = Util.GetConnectionString();
            var ds = new DataSet();
            Util.FillTable(ds, "Customers", conn);

            flexGrid.DataSource = ds;
            flexGrid.DataMember = "Customers";
            flexGrid.RowDetailProvider = (g, r) => new C1InputPanelRowDetail();
            flexGrid.RowDetailsVisibilityMode = RowDetailsVisibilityMode.VisibleWhenSelected;
        }
    }
}

File: NetFramework/FlexGrid/VB/RowDetails/Samples/InputPanelSample.vb
Imports C1.Win.FlexGrid

Public Class InputPanelSample
    Private Sub InputPanelSample_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim conn As String = Util.GetConnectionString()
        Dim ds = New DataSet()
        Util.FillTable(ds, "Customers", conn)

        flexGrid.DataSource = ds
        flexGrid.DataMember = "Customers"

        flexGrid.RowDetailProvider = Function(g, r) New C1InputPanelRowDetail()
        flexGrid.RowDetailsVisibilityMode = RowDetailsVisibilityMode.VisibleWhenSelected
    End Sub
End Class
